packages/radix-ui-themes/src/components/dialog.tsx

```diff
@@ -41,11 +41,15 @@ const DialogContent = React.forwardRef<DialogContentElement, DialogContentProps>
       <DialogPrimitive.Portal container={container} forceMount={forceMount}>
         <Theme asChild>
           <DialogPrimitive.Overlay className="rt-DialogOverlay">
-            <DialogPrimitive.Content
-              {...contentProps}
-              ref={forwardedRef}
-              className={classNames('rt-DialogContent', className)}
-            />
+            <div className="rt-DialogScroll">
+              <div className="rt-DialogScrollPadding">
+                <DialogPrimitive.Content
+                  {...contentProps}
+                  ref={forwardedRef}
+                  className={classNames('rt-DialogContent', className)}
+                />
+              </div>
+            </div>
```

Comment:
Instead of 2 DOM elements, we could achieve it with just 1 more DOM element, or potentially with no extra DOM elements if we distributed some styles onto the pseudo elements, however I found that it adds a layer of indirection in the styles and when inspecting the resulting DOM that I wasn't sure worth it.